ebmdatalab/generate-ebmdatalab-stats.py

```diff
@@ -35,6 +37,64 @@ def extract_citekey(results_url):
         citekey = short_doi_url.replace('https://doi.org', 'doi:10')
     return citekey
 
+def assign_ISO(countries):
+    """
+    Match country names with ISO codes
+    Input: series of country names
+    Returns: dictionary of matches
+    :type countries: pd.Series
+    """
+
+    # Need to hard code a few countries that aren't registered using standard names, so
+    # initializing the country_codes database with these irregular values
+    country_codes = {"South Korea": "KOR", "Democratic Republic of Congo": "COD",
+                     "Democratic Republic of the Congo": "COD"}
+
+    # Identify the most likely 3-letter ISO code for each country
+    failed_matches = list()
+    for country in countries:
+        if country not in country_codes.keys():
+            try:
+                hit = pycountry.countries.get(name=country)
+                if isinstance(hit, type(None)):
+                    # .get retrieves data as class Country
+                    # if it can't find a match, try alternative methods
+                    # .search_fuzzy matching returns a list
+                    hit = pycountry.countries.search_fuzzy(country)
+                    if len(hit) > 1:
+                        hit = pycountry.countries.search_fuzzy(country + ",")
+                    elif isinstance(hit, type(None)):
+                        hit = pycountry.countries.get(official_name=country)
+            except LookupError:
```

Comment:
There are two get calls here - does it matter which one raises the lookup error? I'm imagining that search_fuzzy wouldn't raise one form its name.

Reply:
It is only the `pycountry.countries.get()` command that throws this error, `.search_fuzzy()` I believe returns an empty list. Should this section be reformatted to better reflect that? I wasn't quite sure the best way to handle this many levels of possible alternative strategies, although in retrospect I could just make a function that would return a value whenever it first succeeds.

Reply:
I noticed there are two .get calls, and if you hit this error the first one you will never do the fuzzy search. We also won't know which .get call failed. I'm not sure that either of these matter, so I wanted to raise them to see if it would make a difference.